qiskit_experiments/library/randomized_benchmarking/interleaved_rb_analysis.py

```diff
@@ -198,3 +199,43 @@ def _post_analysis(self, result_data: CurveAnalysisResultData) -> CurveAnalysisR
         result_data["EPC_systematic_bounds"] = [max(systematic_err_l, 0), systematic_err_r]
 
         return result_data
+
+    def _run_analysis(self, experiment_data, **options):
+        """Run analysis on circuit data."""
+
+        analysis_results, figures = super()._run_analysis(experiment_data, **options)
+
+        # Manual formatting for analysis result
+        # This sort of post-processing should be refactored into CurveAnalysis
+        # so that it works with the AnalysisResult dataclasses
+        curve_result = analysis_results[0]
+        chisq = curve_result.chisq
+        quality = curve_result.quality
+        result_data = curve_result.extra
+
+        alpha = FitVal(get_opt_value(result_data, "alpha"), get_opt_error(result_data, "alpha"))
+        analysis_results.append(AnalysisResultData("alpha", alpha, chisq=chisq, quality=quality))
+
+        alpha_c = FitVal(
+            get_opt_value(result_data, "alpha_c"), get_opt_error(result_data, "alpha_c")
+        )
+        analysis_results.append(
+            AnalysisResultData("alpha_c", alpha_c, chisq=chisq, quality=quality)
+        )
+
+        if "EPC" in result_data:
```

Comment:
I think this should be written in post analysis (this is why I prepared the method) so that `_run_analysis` is not overridden (actually post analysis is originally intended to format the result into `AnalysisResultData`). If this doesn't work my class design was not good.

Reply:
Yeah I think this needs some refactoring of CurveAnalysis, which is currently passing around CurveAnalysisResultData instead of analysis results which we should do as follow up